tests/integration/test_ingress_relation.py

```diff
@@ -8,69 +8,109 @@
 import pathlib
 import textwrap
 
+import pytest
 import requests
 from juju.model import Model
 
 
-async def test_ingress_relation(
-    model: Model, deploy_any_charm, run_action, build_and_deploy_ingress
-):
-    """
-    assert: None
-    action: Build and deploy nginx-ingress-integrator charm, also deploy and relate an any-charm
-        application with ingress relation for test purposes.
-    assert: HTTP request should be forwarded to the application.
-    """
-    any_charm_py = textwrap.dedent(
-        """\
-    import pathlib
-    import subprocess
-    import ops
-    from any_charm_base import AnyCharmBase
-    from ingress import IngressPerAppRequirer
-    class AnyCharm(AnyCharmBase):
-        def __init__(self, *args, **kwargs):
-            super().__init__(*args, **kwargs)
-            self.ingress = IngressPerAppRequirer(self, port=8080)
-            self.unit.status = ops.BlockedStatus("Waiting for ingress relation")
-            self.framework.observe(
-                self.on.ingress_relation_changed, self._on_ingress_relation_changed
-            )
-
-        def start_server(self):
-            www_dir = pathlib.Path("/tmp/www")
-            www_dir.mkdir(exist_ok=True)
-            file_path = www_dir / "path" / "ok"
-            file_path.parent.mkdir(exist_ok=True)
-            file_path.write_text(str(self.ingress.url))
-            proc_http = subprocess.Popen(
-                ["python3", "-m", "http.server", "-d", www_dir, "8080"],
-                start_new_session=True,
-            )
-
-        def _on_ingress_relation_changed(self, event):
-            self.unit.status = ops.ActiveStatus()
+def make_any_charm_source(strip_prefix: bool = False) -> str:
+    """Generate the source code for the any-charm with ingress relation.
+
+    Args:
+        strip_prefix: Whether to strip the prefix from the URL path.
+
+    Return:
+        str: The source code of the any-charm.
     """
+    file_path_expr = 'www_dir / "testing-any" / "ok"' if not strip_prefix else 'www_dir / "ok"'
```

Comment:
The `testing` is dynamic based on the Juju model name. I think this should be a variable.